File: daily/2026/3471_Find_the_Largest_Almost_Missing_Integer.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int largestInteger(vector<int>& nums, int k) {
        int n = nums.size();
        if (k == n) return *max_element(nums.begin(), nums.end());
        vector<int> cnt(51, 0);
        for (int i : nums) cnt[i]++;
        if (k == 1){
            for (int i = 50; i >= 0; i--){
                if (cnt[i] == 1) return i;
            }
        }
        int res = -1;
        if (cnt[nums[0]] == 1) res = max(res, nums[0]);
        if (cnt[nums[n-1]] == 1) res = max(res, nums[n-1]);
        return res;
    }
};

int main(){
    vector<int> nums = {3,9,2,1,7};
    int k = 3;
    // vector<int> nums = {3,9,7,2,1,7};
    // int k = 4;
    // vector<int> nums = {0,0};
    // int k = 1;
    // vector<int> nums = {0,0};
    // int k = 2;
    // vector<int> nums = {3,1,7,10,0};
    // int k = 1;
    Solution S;
    cout << S.largestInteger(nums, k) << endl;
    return 0;
}

File: daily_python/3471_Find_the_Largest_Almost_Missing_Integer.py
import os
import sys
from typing import List

class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        n = len(nums)
        if k == n:
            return max(nums)
        cnt = [0] * 51
        for i in nums:
            cnt[i] += 1
        if k == 1:
            for i in range(50, -1, -1):
                if (cnt[i] == 1):
                    return i
        res = -1
        if cnt[nums[0]] == 1:
            res = max(res, nums[0])
        if cnt[nums[-1]] == 1:
            res = max(res, nums[-1])
        return res

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            k = int(lines[1].split('=')[1].strip()[:-1])
            dataset.append((nums, k))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums, k in dataset:
        results.append(solution.largestInteger(nums, k))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')